parsing/builtin_attributes.ml

```diff
@@ -15,6 +15,72 @@
 
 open Asttypes
 open Parsetree
+open Ast_helper
+
+
+module Attribute_table = Hashtbl.Make (struct
+  type t = string with_loc
+
+  let hash : t -> int = Hashtbl.hash
+  let equal : t -> t -> bool = (=)
+end)
```

Comment:
This seemed like the most efficient choice, since the stdlib lacks a mutable set.  `ppxlib` makes the same choice for [their similar structure](https://github.com/ocaml-ppx/ppxlib/blob/af1756cfcfb6ebabd2ebf2234109cd50fce3f2f8/src/attribute.ml#L307).  But what would you recommend?

Reply:
I was thinking of using a `Attribute_set ref` with the attribute set instantiated with a total version of `attr_order` to avoid the `sort` function. But a hashtbl as mutable set is fine.